tests/integration/ha_tests/test_smoke.py

```diff
@@ -0,0 +1,146 @@
+    host = get_unit_address(ops_test, primary_name)
+    result = requests.get(f"http://{host}:8008/health")
+    assert result.status_code == 200
+
+    storage_id_str = storage_id(ops_test, primary_name)
+
+    # Check if storage exists after application deployed
+    assert await is_storage_exists(ops_test, storage_id_str)
+
+    await ops_test.model.remove_application(APPLICATION_NAME, block_until_done=True, destroy_storage=True)
+
+    # Check if storage removed after application removal
+    assert not await is_storage_exists(ops_test, storage_id_str)
+
+    print(storage_id_str)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_app_force_removal(ops_test: OpsTest, charm: str):
+    # Deploy the charm.
+    await ops_test.model.deploy(
+        charm,
+        application_name=APPLICATION_NAME,
+        num_units=1,
+        series=CHARM_SERIES,
+        config={"profile": "testing"},
+    )
+
+    # Reducing the update status frequency to speed up the triggering of deferred events.
+    await ops_test.model.set_config({"update-status-hook-interval": "10s"})
+
+    await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1500)
+    assert ops_test.model.applications[APPLICATION_NAME].units[0].workload_status == "active"
+
+    primary_name = await get_primary(ops_test, ops_test.model.applications[APPLICATION_NAME].units[0].name)
+    host = get_unit_address(ops_test, primary_name)
+    result = requests.get(f"http://{host}:8008/health")
+    assert result.status_code == 200
+
+    storage_id_str = storage_id(ops_test, primary_name)
+
+    # Check if storage exists after application deployed
+    assert await is_storage_exists(ops_test, storage_id_str)
+
+    await ops_test.model.remove_application(APPLICATION_NAME, block_until_done=True, force=True, no_wait=True)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_app_garbage_ignorance(ops_test: OpsTest, charm: str):
+    # Deploy the charm.
+    await ops_test.model.deploy(
+        charm,
+        application_name=APPLICATION_NAME,
+        num_units=1,
+        series=CHARM_SERIES,
+        config={"profile": "testing"},
+    )
+
+    # Reducing the update status frequency to speed up the triggering of deferred events.
+    await ops_test.model.set_config({"update-status-hook-interval": "10s"})
+
+    await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1500)
+    assert ops_test.model.applications[APPLICATION_NAME].units[0].workload_status == "active"
+
+    primary_name = await get_primary(ops_test, ops_test.model.applications[APPLICATION_NAME].units[0].name)
+    host = get_unit_address(ops_test, primary_name)
+    result = requests.get(f"http://{host}:8008/health")
+    assert result.status_code == 200
+
+    # Create test database to check there is no resouces conflicts
+    await create_db(ops_test, APPLICATION_NAME, TEST_DATABASE_RELATION_NAME)
+
+    # Deploy duplicaate charm
```

Comment:
As discussed on the sync today: we are testing Juju here, but we need to test charm.
Juju guaranties that independent applications will have interdependent storages, no need to test it here, IMHO. But you can keep it as already created as it should be fast.

Our goal is to test charm:
 * charm A can start with storage from A (cleanly). Even if there are some garbage lock files on storage.
 * charm A can start with storage from B (with warning in log, about foreign storage, but deploying anyway as it can be an app recovery process).

P.S. the goal of smoke test is a fast result, so try to keep the test fast (in time).